### DISCUSSION TITLE
TwigLiveComponent - dynamically url path

### DISCUSSION THREAD

**Discussion by disceney** [symfony/ux#2172] [Questions & Answers] [answered]
Hello,

I would like to know if it is possible to dynamically modify the rendering URLs with TwigLiveComponent.

For example, here is a component:

```php
<?php

declare(strict_types=1);

namespace App\Twig\Components\User;

...

#[AsLiveComponent]
final class Notification
{
```

And its usage in a Twig template:

```twig
<twig:User:Notification user="{{ app.user }}"/>
```

When called, here is the URL generated in the network tab of the console:

```text
https://interface.dev/_components/User:Notification
```

Is there a way, either globally or not, to generate custom URLs? For example, to obtain a URL in the following form:

```text
https://interface.dev/live/user/notification
```

Thank you for your help

**Comment by smnandre** [comment, marked as answer]
You can either globally (change the route declared in config/routes/live_component.yaml) or per component

(see https://symfony.com/bundles/ux-live-component/current/index.html#define-another-route-for-your-component )
